### ISSUE TITLE
Loklak search shows no results when server is down

### ISSUE THREAD

**Issue by singhpratyush** [fossasia/loklak_search#431]
When the server is down, the app simply points out that there are no results matching the query, which should not happen.

![screenshot from 2017-06-28 21 22 55](https://user-images.githubusercontent.com/10860278/27647060-0572019a-5c48-11e7-9ad3-63634499076a.png)

Ideal behaviour would be showing a message like `Backend under maintenance. Please try again in some time.` Futher, all views should be disabled when such problem occurs.

We can make a call to `/api/status.json` to check status of backend.

@mariobehling @Achint08 @djmgit @hemantjadon @kavithaenair @SarishiNoHara @SKrPl @vibhcool: What are your views on this?

**Comment by djmgit** [comment]
@singhpratyush  yeah you are right, that should be the ideal case, in that way users will actually be able to know why they are not getting any results.

**Comment by hemantjadon** [comment]
Yup This should be fixed, thanks for pointing this out.
There are some other cases which can be handled, like `503: Too many requests`, `502: Bad Gateway`, `Timeout`. These cases should be handled to provide the important feedback to the user. I will be working on this next.

**Comment by Achint08** [comment]
@singhpratyush Thanks for pointing out.

**Comment by hemantjadon** [comment]
IMO we should make a central state for all the errors 404, 500, 502 etc. All the errors will be linked to this state using reducers and actions. The components which will have to update the UI according to the errors will do so using this state.

**Comment by Achint08** [comment]
@hemantjadon Do we need to show up regarding all the errors? 404 errors I guess never occurs and will be caused by a development issue. Most commonly occurring errors are 502, 503 and these are just because of server issues. We can simply show up `Backend under maintenance. Please try again in some time.` 
Looking at the effects, we generally dispatch fail action for all the errors(502/503/no feeds). We can instead dispatch this action `SearchCompleteFailAction('')` in case of server issue. And show up no results page, if there are no feeds in the response we have received by checking the number of feeds received.
